macosx/Torrent.mm

```diff
@@ -252,7 +266,10 @@ - (NSDictionary*)history
         @"Active" : @(self.active),
         @"WaitToStart" : @(self.waitingToStart),
         @"GroupValue" : @(fGroupValue),
-        @"RemoveWhenFinishSeeding" : @(_removeWhenFinishSeeding)
+        @"RemoveWhenFinishSeeding" : @(_removeWhenFinishSeeding),
+        @"IsMagnet" : @(self.isMagnet),
+        @"MagnetLink" : self.magnetLink,
+        @"MagnetLocation" : (NSString *)self.magnetLocation
```

Comment:
This is going to break if it's an NSNull.

Reply:
I already said this….
But this does in fact work, I tested it.

Reply:
At least on macOS Monterrey it works just fine

Reply:
Hmm, it works because it's cast to an NSString but never explicitly used as an NSString, and it looks like NSNull can be encoded with the other fields in the dictionary. I think the real solution would be to not encode the location if it's nil (and don't use NSNull)

Reply:
I can revert to sending an empty NSString if you prefer. That seems safer and makes the code more readable